sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -298,6 +351,65 @@ public BlobServiceVersion getServiceVersion() {
         return getBlobContainerAsyncClient(containerName).deleteWithResponse(null, context);
     }
 
+    /**
+     * Deletes the specified container in the storage account if it exists. For
+     * more information see the <a href="https://docs.microsoft.com/rest/api/storageservices/delete-container">Azure
+     * Docs</a>.
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <!-- src_embed com.azure.storage.blob.BlobServiceAsyncClient.deleteBlobContainerIfExists#String -->
+     * <pre>
+     * client.deleteBlobContainerIfExists&#40;&quot;containerName&quot;&#41;.subscribe&#40;
+     *     response -&gt; System.out.printf&#40;&quot;Delete container completed%n&quot;&#41;,
+     *     error -&gt; System.out.printf&#40;&quot;Delete container failed: %s%n&quot;, error&#41;&#41;;
+     * </pre>
+     * <!-- end com.azure.storage.blob.BlobServiceAsyncClient.deleteBlobContainerIfExists#String -->
+     *
+     * @param containerName Name of the container to delete
+     * @return A {@link Mono} containing status code and HTTP headers, or null if the container does not exist.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteBlobContainerIfExists(String containerName) {
+        try {
```

Comment:
Only withResponse methods should need the try/catch. This method just calls into the withResponse. Double check this on other clients, too